Tests/Runtime/Session/BacktraceSessionTests.cs

```diff
@@ -0,0 +1,282 @@
+        }
+
+
+        [Test]
+        public void BacktraceSession_ShouldSkipMoreUniqueEventsWhenHitTheLimit_DataWasntSendToBacktrace()
+        {
+            const int maximumNumberOfEvents = 3;
+            const int numberOfTestEvents = 10;
+            var backtraceSession = new BacktraceSession(_attributeProvider, _submissionUrl, 0, maximumNumberOfEvents);
+
+            for (int i = 0; i < numberOfTestEvents; i++)
+            {
+                backtraceSession.AddUniqueEvent($"{UniqueAttributeName} {i}", new Dictionary<string, string>() { { $"{UniqueAttributeName} {i}", "value" } });
+            }
+
+            Assert.AreEqual(maximumNumberOfEvents, backtraceSession.Count());
+        }
+
+        [Test]
+        public void BacktraceSession_ShouldSkipMoreSessionEventsWhenHitTheLimit_DataWasntSendToBacktrace()
+        {
+            const int maximumNumberOfEvents = 3;
+            const int numberOfTestEvents = 10;
+            var backtraceSession = new BacktraceSession(_attributeProvider, _submissionUrl, 0, maximumNumberOfEvents);
+
+            for (int i = 0; i < numberOfTestEvents; i++)
+            {
+                backtraceSession.AddSessionEvent($"{SessionEventName} {i}");
+            }
+
+            Assert.AreEqual(maximumNumberOfEvents, backtraceSession.Count());
+        }
+
+        [Test]
+        public void BacktraceSession_ShouldTriggerDownloadViaTickMethodWhenReachedMaximumNumberOfEvents_DataWasSendToBacktrace()
+        {
+            const int maximumNumberOfEvents = 3;
+            const int defaultTimeIntervalInMs = 10;
+            var backtraceSession = new BacktraceSession(_attributeProvider, _submissionUrl, defaultTimeIntervalInMs, maximumNumberOfEvents);
+            var requestHandler = new BacktraceHttpClientMock();
+            backtraceSession.RequestHandler = requestHandler;
+            for (int i = 0; i < maximumNumberOfEvents; i++)
+            {
+                backtraceSession.AddSessionEvent($"{SessionEventName} {i}");
+            }
+
+            backtraceSession.Tick(defaultTimeIntervalInMs + 1);
+
+            Assert.AreEqual(0, backtraceSession.Count());
+            Assert.AreEqual(1, requestHandler.NumberOfRequests);
+        }
+
+        [Test]
+        public void BacktraceSession_ShouldntTriggerDownloadViaTickMethodWhenDidntReachMaximumNumberOfEvents_DataWasSendToBacktrace()
+        {
+            const int maximumNumberOfEvents = 3;
+            const int expectedNumberOfEvents = 2;
+            var backtraceSession = new BacktraceSession(_attributeProvider, _submissionUrl, 1, maximumNumberOfEvents);
+            var requestHandler = new BacktraceHttpClientMock();
+            backtraceSession.RequestHandler = requestHandler;
+            for (int i = 0; i < expectedNumberOfEvents; i++)
+            {
+                backtraceSession.AddSessionEvent($"{SessionEventName} {i}");
+            }
+
+            backtraceSession.Tick(0);

```

Comment:
Why tick 0?

Reply:
0 means current time - We have value 0 on the application startup. For testing purpose it was enough because we wanted to force upload process when we have enough data to send or (in this example) test if we will send anything when we shouldn't.